cartridges/int_algolia/cartridge/scripts/algolia/model/algoliaProduct.js

```diff
@@ -257,9 +259,12 @@ var aggregatedValueHandlers = {
         return productPrice;
     },
     in_stock: function (product) {
-        // the in_stock property now exports the SFCC attribute product.availabilityModel.inStock,
-        // not whether the threshold defined in Algolia_InStockThreshold value is reached.
-        return product.availabilityModel.inStock;
+        // the in_stock property now exports a boolean indicating whether the number of products in stock
+        // (ATS or available to sell) is higher than the threshold set in the custom site preference Algolia_InStockThreshold,
+        // as originally intended. The previous logic was flawed in that it compared the sitepref to product.availabilityModel.availability,
+        // which is a number between 0 and 1 and does not represent the number of products in stock.
+        let inventoryRecord = product.getAvailabilityModel().getInventoryRecord(); // can be null
+        return (inventoryRecord ? inventoryRecord.getATS().getValue() > ALGOLIA_IN_STOCK_THRESHOLD : false);
```

Comment:
The doc says
> If the stock of a product is lower than this value

So it should be
```suggestion
        return (inventoryRecord ? inventoryRecord.getATS().getValue() >= ALGOLIA_IN_STOCK_THRESHOLD : false);
```